src/components/Filters/hooks/use-filters.ts

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+
+import type {FilterValue, FiltersValue} from '../Filters';
+
+export type FiltersReducerState = Record<string, any>;
+
+/**
+ * This is a plug-n-play approach to use Filters component.
+ *
+ * With this hook you don't need to implement state management for filters component. This might be useful
+ * if you do not need any extra logic before/after applying filters. Otherwise, you should implement those handlers by yourself
+ * and pass them to Filters component.
+ *
+ * @param initialValues Initial values, e.g. previous filters values restored from localStorage.
+ * @param emptyValue Empty values, by default equals to initialValues. This property will be used to set filtersValue after resetting them.
+ */
+export function useFilters<T extends Record<string, any> = FiltersValue>(
+    initialValues: T,
+    emptyValue: T = initialValues,
+) {
+    const [filtersValues, setFiltersValues] = React.useState<FiltersValue>(initialValues);
+
+    const handleChangeFilterValue = React.useCallback(
+        (id: string, value: FilterValue) =>
+            setFiltersValues((currentValue) => ({
+                ...currentValue,
+                [id]: value,
+            })),
+        [setFiltersValues],
+    );
+
+    const handleRemoveFilterValue = React.useCallback(
+        (id: string, _values: FilterValue) =>
```

Comment:
Here and below: why we need `_values` definition here?

Reply:
removed